micro-rdk/src/esp32/entry.rs

```diff
@@ -337,55 +369,105 @@ async fn serve_async_with_external_network<S>(
     exec: Esp32Executor,
     info: Option<ProvisioningInfo>,
     storage: S,
-    repr: RobotRepresentation,
+    mut repr: RobotRepresentation,
     network: impl Network,
     max_webrtc_connection: usize,
 ) -> Result<(), Box<dyn std::error::Error>>
 where
-    S: RobotCredentialStorage + WifiCredentialStorage + Clone + 'static,
-    <S as RobotCredentialStorage>::Error: Debug,
-    ServerError: From<<S as RobotCredentialStorage>::Error>,
+    S: RobotConfigurationStorage + WifiCredentialStorage + Clone + 'static,
+    <S as RobotConfigurationStorage>::Error: Debug,
+    ServerError: From<<S as RobotConfigurationStorage>::Error>,
     <S as WifiCredentialStorage>::Error: Sync + Send + 'static,
 {
     use crate::common::provisioning::server::serve_provisioning_async;
 
     use super::conn::mdns::Esp32Mdns;
 
+    let mut client_connector = Esp32TLS::new_client();
     let info = info.unwrap_or_default();
     let mut last_error: Option<Box<dyn std::error::Error>> = None;
 
-    loop {
-        // Credentials are present let's check we can connect
-        if storage.has_stored_credentials() {
-            let validated = loop {
-                // should check internet when implementing Cached Config
-                if let Err(error) = network.is_connected() {
-                    log::info!(
-                        "Externally managed network, not connected yet cause {:?}",
-                        error
-                    );
-                    Timer::after(Duration::from_secs(3)).await;
-                    continue;
+    let app_client = 'provisioned: loop {
+        if storage.has_robot_credentials() {
+            log::info!("Found cached robot credentials; attempting to serve");
+
+            if storage.has_robot_configuration() {
```

Comment:
nit: merge both if with &&

Reply:
There is logic further down that executes in the has_robot_credentials case independently of has_robot_configuration, so I don't think they can be combined. See the code starting at `let mut Duration = ...`.